include/devclass.h

```diff
@@ -0,0 +1,26 @@
+#ifndef _DEVCLASS_H_
+#define _DEVCLASS_H_
+
+#include <device.h>
+#include <queue.h>
+
+typedef struct devclass devclass_t;
+
+// should be exported to some toplevel, machine independent file
+// SET_DECLARE(devclasses, devclass_t);
+
+struct devclass {
+  const char *name;
+  driver_t **drivers;
+};
+
+#define DEVCLASS_CREATE(dc_name)                                               \
+  SET_DECLARE(dc_name, driver_t);                                              \
+  devclass_t dc_name##_devclass = {.name = (#dc_name),                         \
+                                   .drivers = SET_BEGIN(dc_name)};             \
+  SET_ENTRY(devclasses, dc_name##_devclass);
+
+#define DEVCLASS_ADD(dc_name, driver_structure)                                \
```

Comment:
what about renaming it to DEVCLASS_ADD_DRV ? Having both DEVCLASS_CREATE and DEVCLASS_ADD is confusing, I imagine I would need to check in code each time to distuingush them

Reply:
Good idea. I'll do that.